common/lib/xmodule/xmodule/vertical_block.py

```diff
@@ -78,7 +78,7 @@ def _student_or_public_view(self, context, view):
         # pylint: disable=no-member
         for child in child_blocks:
             child_block_context = copy(child_context)
-            if child in child_blocks_to_complete_on_view:
+            if child_blocks_to_complete_on_view and child in child_blocks_to_complete_on_view:
```

Comment:
This fixes the test but the underlying issue is that child is SplitTestDescriptorWithMixins type object which is not hashable. So we can not compare it with child_blocks_to_complete_on_view which is set. So we will need to define the hash value for this class too. so that we could compare both of them but was not able to find that class in platform.

Reply:
@feanil we are facing this issue on some new issues too.

Reply:
@Ayub-Khan: Is it possible to examine usage keys instead of blocks here?

Reply:
Equality/hashing of XModules/XBlocks is more complex, and I'd rather avoid delving into that if at all possible.

Reply:
@ormsbee what is your opinion regarding this fix https://github.com/edx/edx-platform/pull/21641/files#diff-7fa42f330b2324628ef0cb3e114649b7R81. if we type cast we can compare and wont need to make any other changes.